src/score_hv/harvesters/gsi_satellite_radiance.py

```diff
@@ -0,0 +1,415 @@
+                    if var == 'var':
+                        value = float(line2list[2].split()[0])
+                    elif var == 'varch_cld':
+                        value = float(line2list[3].split()[0])
+                    elif var == 'use':
+                        value = int(line2list[4].split()[0])
+                    elif var == 'ermax':
+                        value = float(line2list[5].split()[0])
+                    elif var == 'b_rad':
+                        value = float(line2list[6].split()[0])
+                    elif var == 'pg_rad':
+                        value = float(line2list[7].split()[0])
+                    elif var == 'icld_det':
+                        value = int(line2list[8].split()[0])
+                    elif var == 'icloud':
+                        value = int(line2list[9].split()[0])
+                    elif var == 'iaeros':
+                        value = int(line2list[10].split()[0])
+                    elif var == BIAS_CORR_COEF_STR:
+                        value = list() # empty list for bias
+                                       # correction coeficients
+                    
+                    data_usage_dict[var] = value    
+                
+                self.channels[series_number] = {
+                    'observation_type': observation_type,
+                    'channel': channel,
+                    'data_usage_dict': data_usage_dict
+                }
+                
+                # store channel numbers
+                if observation_type in self.obs_type_channels.keys():
+                    self.obs_type_channels[observation_type].append(
+                        channel)
+                else:
+                    self.obs_type_channels[observation_type] = [channel]
+                    
+                # store series numbers
+                if observation_type in self.obs_type_series_numbers.keys():
+                    self.obs_type_series_numbers[observation_type].append(
+                        series_number)
+                else:
+                    self.obs_type_series_numbers[observation_type] = [
+                        series_number]
+                
+                if series_number == self.nchannels:
+                    """this is the last channel
+                    """
+                    radinfo_read1 = False
+                
+            elif radinfo_read2:
+                """RADINFO_READ PART 2
+                
+                harvest bias correction coeficients
+                """
+
+                series_number = int(line_parts[0])
+                #channel_index = series_number - 1
+                
+                # partial assurance that this is the correct channel, by name
+                assert line_parts[1] == self.channels[series_number]['observation_type']
+                bias_corr_coef = line_parts[2:]
+                
+                if len(bias_corr_coef) == N_BIAS_CORR_COEF: # there should always be 12 channels
```

Comment:
why should there always be 12 channels? I thought the point was that sats can have a lot of different channels??

Reply:
adding on, that if this is not referring to satellite channels but number of bias correction coefficient columns then it probably should be changed to a more specific statement to avoid confusion between usage of the word 'channel'

Reply:
yes, thank you and corrected